### PR TITLE
API to get node count in ring

### PR DESCRIPTION
**Pull request by jpmelos**
API to return number of active nodes in the ring.

### CODE REVIEW

File: redis_hashring/__init__.py [hunk 1]
```diff
@@ -285,6 +292,14 @@ def update(self):
                 else:
                     self.ranges.append((0, RING_SIZE))
 
+        self.node_count = len(nodes)
+
+    def get_node_count(self):
```
Comment (thomasst) [inline redis_hashring/__init__.py:297]: My opinion would be to rename the variable to `_node_account` if we're not supposed to directly access the variable, or else just remove the getter.

**Review by thomasst — APPROVED** [review]

File: redis_hashring/__init__.py (hunk 1 above)
Comment (jpmelos) [reply redis_hashring/__init__.py:297]: It makes sense to me! There's a lot of other attributes in that class that could also be private too, so I made that happen in https://github.com/closeio/redis-hashring/pull/25/commits/c7231cea3d65c8d326fd0cd186c519d094ba3bd2. Let me know if that doesn't make sense to you.

Also, there's a lot of methods that are publicly exposed now, and I'm wondering if they should continue to be public... Things like `poll`, `heartbeat`, `update`, `remove`, `cleanup`, etc... But I guess it makes sense for those to be public. That way, if someone wants to use this library with something other than threads and `gevent`, they can implement it with a different concurrency framework. Makes sense?

Comment (jpmelos) [reply redis_hashring/__init__.py:297]: Well, I guess how ring points are calculated is an implementation detail and there's no need to expose it, right? I'm making `contains_ring_point` and `key_as_ring_point`. Let me know if there's a good reason to expose those.

Comment (thomasst) [reply redis_hashring/__init__.py:297]: Sure. No need to overthink it. Things like `conn` usually don't need to be accessed externally but sometimes I've found myself needing access to underlying connections (in the general case, not specifically in the hashring). In this particular case I just didn't want to have both a getter and also a non-underscored variable.

File: redis_hashring/__init__.py [hunk 2]
```diff
@@ -115,9 +115,11 @@ def __init__(self, conn, key, n_replicas=RING_REPLICAS):
             for _ in range(n_replicas)
         ]
 
+        # Number of nodes currently active in the ring.
+        self._node_count = 0
         # List of tuples of ranges this node is responsible for, where a tuple
         # (a, b) includes any N matching a <= N < b.
-        self.ranges = []
+        self._ranges = []
```
Comment (thomasst) [inline redis_hashring/__init__.py:122]: We actually make use of this.

**Pull request merged**